spark/src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -134,7 +134,12 @@ case class CreateDeltaTableCommand(
     }
 
     val tableLocation = getDeltaTablePath(tableWithLocation)
-    val deltaLog = DeltaLog.forTable(sparkSession, tableLocation)
+    // To be safe, here we only extract file system options from table storage properties, to create
+    // the DeltaLog.
+    val fileSystemOptions = table.storage.properties.filter { case (k, _) =>
+      DeltaTableUtils.validDeltaTableHadoopPrefixes.exists(k.startsWith)
+    }
+    val deltaLog = DeltaLog.forTable(sparkSession, tableLocation, fileSystemOptions)
```

Comment:
Can we add a test for this? that the right file system options are passed to DeltaLog.

Reply:
I may need some help on it. Do we have a fake file system in the Delta test framework? This is the write options and I can't verify it after the Delta table is created, as write options are not persisted into the Delta log.

Reply:
Yes there is a FakeFileSystem and here is a [reference test suite using it](https://github.com/delta-io/delta/blame/master/spark/src/test/scala/io/delta/tables/DeltaTableSuite.scala#L290).